python/teamsai/prompts/prompt_manager.py

```diff
@@ -0,0 +1,69 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+import abc
+from typing import TypeVar, Generic, Any, Awaitable, Union
+from botbuilder.core import TurnContext
+from ..turn_state import TurnState
+from .prompt_template import PromptTemplate
+
+T = TypeVar("T", TurnState)
+
+
+class PromptManager(abc.ABC, Generic[T]):
```

Comment:
We may use Protocol instead of ABC

Reply:
ok ya that makes sense to me I'll update it with my next PR that will be more focused on these pieces of code